cmake/third_party/openfst.cmake

```diff
@@ -14,12 +14,74 @@ if(NOT openfst_POPULATED)
 
     add_subdirectory(${openfst_SOURCE_DIR} ${openfst_BINARY_DIR})
 
-    install(DIRECTORY ${openfst_SOURCE_DIR}/src/include/ DESTINATION include/
-            FILES_MATCHING PATTERN "*.h")
-
-    install(TARGETS fst
-            EXPORT kaldi-targets
-            ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
-            LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
-            RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
+
+    if(CONDA_ROOT)
+
+        if(MSVC)
+            install(TARGETS fstarcsort fstclosure fstcompile fstcompose fstconcat fstconnect fstconvert fstdeterminize
+                            fstdifference fstdisambiguate fstdraw fstencode fstepsnormalize fstequal fstequivalent
+                            fstinfo fstintersect fstinvert fstisomorphic fstmap fstminimize fstprint fstproject fstprune
+                            fstpush fstrandgen fstrelabel fstreplace fstreverse fstreweight fstrmepsilon fstshortestdistance
+                            fstshortestpath fstsymbols fstsynchronize fsttopsort fstunion
+            RUNTIME
+                DESTINATION ${CMAKE_INSTALL_BINDIR}
+                COMPONENT kaldi
+            )
+            install(TARGETS farcompilestrings farcreate farequal farextract farinfo farisomorphic farprintstrings
+            RUNTIME
+                DESTINATION ${CMAKE_INSTALL_BINDIR}
+                COMPONENT kaldi
+            )
+            install(TARGETS fstlinear fstloglinearapply
+            RUNTIME
+                DESTINATION ${CMAKE_INSTALL_BINDIR}
+                COMPONENT kaldi
+            )
+            install(TARGETS mpdtcompose mpdtexpand mpdtinfo mpdtreverse
+            RUNTIME
+                DESTINATION ${CMAKE_INSTALL_BINDIR}
+                COMPONENT kaldi
+            )
+            install(TARGETS pdtcompose pdtexpand pdtinfo pdtreplace pdtreverse pdtshortestpath
+            RUNTIME
+                DESTINATION ${CMAKE_INSTALL_BINDIR}
+                COMPONENT kaldi
+            )
+            install(TARGETS fstspecial
+            RUNTIME
+                DESTINATION ${CMAKE_INSTALL_BINDIR}
+                COMPONENT kaldi
+            )
+
+            install(DIRECTORY ${openfst_SOURCE_DIR}/src/include/ DESTINATION include/
+                    FILES_MATCHING PATTERN "*.h")
+
+            install(TARGETS fst
+                    EXPORT kaldi-targets
+                    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT kaldi
+                    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT kaldi NAMELINK_SKIP
+                    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} COMPONENT kaldi)
+        else()
+
+            install(TARGETS fst
+                    LIBRARY
+                        DESTINATION ${CMAKE_INSTALL_LIBDIR}
+                        COMPONENT kaldi
+                        NAMELINK_SKIP # Link to so with version to avoid conflicts with OpenFst 1.8.1 on conda
+                    )
+
+        endif()
```

Comment:
I'll investigate which of these binaries are in fact required.

BTW, you can build OpenFST binaries against a static OpenFST library. The libfst in fact has a tiny footprint, as the most of its magic is based on C++ templates and is thus header-only. I usually do exactly that. This will divorce the FST library used to build Kaldi from the one baked into the binaries. Building Kaldi fully static results in ~15GB of binaries, so probably not the best approach for packaging, but its required fst\* and far\* binaries subset isn't very large, and isn't reduced much by linking against a shared SO library, as it's really very small. You need to build the static library either using PIC, which is a much smaller performance hit on x64 than it used to be on x86, or use the large/huge memory model, so that all relative jump instructions are 64 bit in side. Doing neither may result in the linker complaining of being unable to fit the jmp offset into the space left in the binary by the compiler. Doing neither will likely result in linker computing a relocation overflowing the one generated by the compiler for the small memory model, of type `R_X86_64_32`, with an error message like "relocation truncated to fit: R_X86_64_32". The large memory model forces the compiler to generate relocations of type R_X86_64_64, not overflowable in a 64-bit address space, resulting in somewhat larger code; PIC will add an indirection for calls, but it is, again, handled much better by the 64-bit mode of CPU than it used to be in the 32-bit mode. Both approaches have a comparable speed, huge relocs beating PIC but only marginally, due to their negative impact on the L1C cache.